### PR TITLE
Us112471/question

### PR DESCRIPTION
**Pull request by jwong-d2l**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
     },
     "frauLocalAppResolver": {
       "appClass": "urn:d2l:fra:class:continuous-professional-development",
-      "dist": "./build/dev"
+      "dist": "./build/dev",
+      "hostname": "localhost"
```
Comment (jwong-d2l) [inline package.json]: this is fine because it's only for local

File: polymer.json [hunk 2]
```diff
@@ -10,6 +10,10 @@
     "package.json",
     "index.html"
   ],
+  "lint": {
+    "rules": [],
+    "ignoreWarnings": ["non-literal-import"]
```
Comment (jwong-d2l) [inline polymer.json]: this ignores ben's thing

File: src/services/cpd-service.js [hunk 3]
```diff
@@ -12,21 +12,68 @@ d2lfetch.use({
 });
 
 export class CpdService {
+
 	static CpdPath(action) { return `/d2l/api/customization/cpd/1.0/${action}`; }
+
+	static get Create() {
```
Comment (jwong-d2l) [inline src/services/cpd-service.js]: This is the beginnings of my generic create stuff.  It's not used in this PR, but could be in the future for subjects and methods.

File: src/helpers/cpdRoutes.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ export class CpdRoutes {
 	static get Pending() { return 'pending'; }
 	static get Progress() { return 'target/progress'; }
 	static get Question() { return 'question'; }
+	static QuestionId(questionId) { return `question/${questionId}`; }
```
Comment (TomMichielsD2L) [inline src/helpers/cpdRoutes.js]: We probably also need a MethodId, SubjectId, MethodSortOrder, QuestionSortOrder and SubjectSortOrder

File: src/services/cpd-service.js [hunk 5]
```diff
@@ -12,21 +12,68 @@ d2lfetch.use({
...
+		const request = new Request(`${this.Host}${this.CpdPath(CpdRoutes.Method)}/${methodId}`, {
+			method: 'DELETE',
+			headers: {
+				'Content-Type' : 'application/json'
+			}
+		});
+		return d2lfetch.fetch(request);
+	}
+	static deleteQuestion(questionId) {
+		const request = new Request(`${this.Host}${this.CpdPath(CpdRoutes.Question)}/${questionId}`, {
```
Comment (TomMichielsD2L) [inline src/services/cpd-service.js]:
```suggestion
		const request = new Request(`${this.Host}${this.CpdPath(CpdRoutes.QuestionId(questionId)}`, {
```

File: src/components/cpd-admin-questions.js [hunk 6]
```diff
@@ -1,12 +1,120 @@
...
+		this.cpdService.deleteQuestion(questionId)
+			.then(() => this.fetchQuestions());
+	}
+	fetchQuestions() {
+		return this.cpdService.getQuestions().then(data => this.questions = data);
+	}
+	openEditDialog(e) {
+		const item = e.target.getAttribute('item-json') &&
+			JSON.parse(e.target.getAttribute('item-json')) || {};
+		this.objectData = item;
```
Comment (TomMichielsD2L) [inline src/components/cpd-admin-questions.js]: why not combine?

File: src/components/cpd-admin-questions.js [hunk 7]
```diff
@@ -1,12 +1,120 @@
...
+			li {
+				display: grid;
+				grid-template-columns: 30px auto;
+			}
+			`
+		];
+	}
+	constructor() {
+		super();
+		this.cpdService = CpdServiceFactory.getCpdService();
```
Comment (TomMichielsD2L) [inline src/components/cpd-admin-questions.js]: At what point should we just put this in the BaseMixin?

File: src/services/demo-cpd-service.js [hunk 8]
```diff
@@ -6,12 +6,20 @@ const getOptions = {
...
+
 export class DemoCpdService {
 
 	static createRecord() {}
 
+	static createSubject() {}
+
 	static deleteRecord() {}
 
+	static deleteSubject(subjectId) {
```
Comment (TomMichielsD2L) [inline src/services/demo-cpd-service.js]: I think we missed a few functions in this service :)

File: src/components/cpd-admin-questions.js (hunk 7 above)
Comment (jwong-d2l) [reply src/components/cpd-admin-questions.js]: are there any places where we use localize but not the service?

File: src/helpers/cpdRoutes.js (hunk 4 above)
Comment (jwong-d2l) [reply src/helpers/cpdRoutes.js]: I didn't want to touch method and subject yet.  I do want to deal with sortorder probably.

File: src/components/cpd-admin-questions.js (hunk 7 above)
Comment (TomMichielsD2L) [reply src/components/cpd-admin-questions.js]: Not sure, it just feels like we have this line of code almost everywhere

Comment (BraydonKains) [reply src/components/cpd-admin-questions.js]: A couple of the small utility components. But it wouldn't be a huge detriment to just move this to the BaseMixin, I'd support that.

**Pull request merged**